locals.tf

```diff
@@ -124,7 +121,8 @@ EOF
   vgw_container = {
     name = "envoy"
     # See README.md or https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html for latest version
-    image_tag = length(var.envoy_proxy_image) > 0 ? var.envoy_proxy_image : "public.ecr.aws/appmesh/aws-appmesh-envoy:v1.29.6.0-prod"
+    # For private ECS, would need private endpoint for ECR to pull the image
+    image_tag = length(var.envoy_proxy_image) > 0 ? var.envoy_proxy_image : "840364872350.dkr.ecr.${var.region}.amazonaws.com/aws-appmesh-envoy:v1.29.6.1-prod"
```

Comment:
Non-blocking suggestion: changing this default container image away from the `public.ecr.aws` address to an explicit regional ECR makes sense from an optimization standpoint, but this takes some information with it: now someone reading this code has to work backwards from the container image to discover that this is an AWS-provided image. It might be worthwhile to call out this particular ECR is provided by AWS rather than Launch either in a comment here or as part of the variable description so that there's no questions about from where our default is being sourced.

Reply:
added the comment with reason